liquidctl/driver/kraken_three_x.py

```diff
@@ -0,0 +1,327 @@
+"""liquidctl driver for Kraken X3 devices from NZXT.
+
+ Supported devices
+ -----------------
+
+  - [ ] NZXT Kraken X53
+  - [ ] NZXT Kraken X63
+  - [✓] NZXT Kraken X73
+
+ Supported features
+ ------------------
+
+  - [✓] general monitoring
+  - [✓] pump speed control
+  - [✓] lighting control
+  - [✓] lighting control advanced - super-breathing, super-fixed, wings, water-cooler
+  - [✓] pump speed curve
+
+ ---
+
+ liquidctl driver for Kraken X3 devices from NZXT.
+ Copyright (C) 2020–2020  Jonas Malaco
+ Copyright (C) 2020–2020  Tom Frey
+ Copyright (C) 2020–2020  each contribution's author
+
+ This file is part of liquidctl.
+
+ liquidctl is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ liquidctl is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ """
+
+import logging
+import itertools
+
+from liquidctl.driver.usb import UsbHidDriver
+from liquidctl.util import normalize_profile, interpolate_profile
+
+LOGGER = logging.getLogger(__name__)
+
+_READ_LENGTH = 64
+_WRITE_LENGTH = 64
+_MAX_READ_ATTEMPTS = 12
+_CRITICAL_TEMPERATURE = 59
+
+_COLOR_CHANNELS = {
+    'ring': 0x02,
+    'logo': 0x04,
+}
```

Comment:
I recently learned that channel IDs on HUE 2 devices are sent as a bit mask.

So if ring is `0b010` and logo is `0b100`, I assume `0b001` is the channel for accessories connected to the HUE 2 header.  Maybe name that channel "accessory".

You can also add a "sync" `0b111` pseudo-channel (which you can use to test that, like other HUE 2 devices, this is indeed a bit mask and that setting both at once works).

Reply:
Using @wamphlett 's captures I confirmed that the "accessory" channel has id 0b001.

Not sure if "accessory" is a great name though: it's long and possibly prone to typos by non-English speakers.

Reply:
d1a2117